pyquil/api/_qpu.py

```diff
@@ -154,3 +182,63 @@ def _build_patch_values(self) -> dict:
             patch_table[k.name][k.index] = v
 
         return patch_table
+
+    def _build_arithmetic_patch_values(self) -> str:
+        """
+        Update the variables shim with the final values to be patched into the gate parameters,
+        according to the arithmetic expressions in the original program.
+
+        For example:
+
+            DECLARE theta REAL
+            RZ(3 * theta) 0
```

Comment:
an example with two (or more) things-we-multiply-theta-by would make this more instructive